src/etc/inc/interfaces.inc

```diff
@@ -1391,6 +1327,130 @@ function interface_gif_configure(&$gif, $gifkey = "") {
 	return $gifif;
 }
 
+/* NOTE: $vxlankey is not used but useful for passing this function to array_walk. */
+function interface_vxlan_configure(&$vxlan, $vxlankey = "") {
+	global $config, $g;
+
+	if (!is_array($vxlan)) {
+		return -1;
+	}
+
+	$realif = convert_friendly_interface_to_real_interface_name($vxlan['if']);
+	if (!interface_is_vlan($realif)) {
+		$realif = get_real_interface($vxlan['if']);
+	}
+	$realifip = get_interface_ip($vxlan['if']);
+	$realifip6 = get_interface_ipv6($vxlan['if']);
+
+	/* make sure the parent interface is up */
+	interfaces_bring_up($realif);
+
+	if (platform_booting() || !(empty($vxlan['vxlanif']))) {
+		pfSense_interface_destroy($vxlan['vxlanif']);
+		pfSense_interface_create($vxlan['vxlanif']);
+		$vxlanif = $vxlan['vxlanif'];
+	} else {
+		$vxlanif = pfSense_interface_create("vxlan");
+	}
+
+	$ifconfigcmd = "/sbin/ifconfig {$vxlanif} vxlanid {$vxlan['vxlanid']} vxlandev {$realif} ";
+
+	if (is_mcast($vxlan['remote-addr'])) {
+		$ifconfigcmd .= "vxlangroup {$vxlan['remote-addr']} ";
+	} else {
+		$ifconfigcmd .= "vxlanremote {$vxlan['remote-addr']} ";
+	}
+
+	if (is_ipaddrv6($vxlan['remote-addr'])) {
+		if ($realifip6) {
+			$ifconfigcmd .= "vxlanlocal {$realifip6} ";
+		} else {
+			return;
+		}
+	} else {
+		if ($realifip) {
+			$ifconfigcmd .= "vxlanlocal {$realifip} ";
+		} else {
+			return;
+		}
+	}
+
+	if (!empty($vxlan['vxlanlocalport'])) {
+		$ifconfigcmd .= "vxlanlocalport {$vxlan['vxlanlocalport']} ";
+	}
+
+	if (!empty($vxlan['vxlanremoteport'])) {
+		$ifconfigcmd .= "vxlanremoteport {$vxlan['vxlanremoteport']} ";
+	}
+
+	if (!empty($vxlan['vxlanttl'])) {
+		$ifconfigcmd .= "vxlanttl {$vxlan['vxlanttl']} ";
```

Comment:
Needs `escapeshellarg()` around variables used in commands.

Reply:
All fixed